### PR TITLE
[VEN-2715] WeETHAccountantOracle Oracle

### PR DESCRIPTION
**Pull request by web3rover**
## Description

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/oracles/WeETHAccountantOracle.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+pragma solidity 0.8.25;
+
+import { CorrelatedTokenOracle } from "./common/CorrelatedTokenOracle.sol";
+import { IAccountant } from "../interfaces/IAccountant.sol";
+import { ensureNonzeroAddress } from "@venusprotocol/solidity-utilities/contracts/validators.sol";
+
+/**
+ * @title WeETHAccountantOracle
+ * @author Venus
+ * @notice This oracle fetches the price of weETH LRT
```
Comment (chechu) [inline contracts/oracles/WeETHAccountantOracle.sol:11]:
```suggestion
 * @notice This oracle fetches the price of Ether.fi tokens based on an `Accountant` contract (i.e. weETHs and weETHk)
```

File: contracts/oracles/WeETHAccountantOracle.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /// @notice Address of Accountant
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IAccountant public immutable ACCOUNTANT;
+
+    /// @notice Constructor for the implementation contract.
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address accountant,
+        address weethLRT,
+        address eth,
```
Comment (chechu) [inline contracts/oracles/WeETHAccountantOracle.sol:23]:
```suggestion
        address weth,
```

File: contracts/oracles/WeETHAccountantOracle.sol [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        address resilientOracle
+    ) CorrelatedTokenOracle(weethLRT, eth, resilientOracle) {
```
Comment (chechu) [inline contracts/oracles/WeETHAccountantOracle.sol:25]:
```suggestion
    ) CorrelatedTokenOracle(weethLRT, weth, resilientOracle) {
```

File: contracts/oracles/WeETHAccountantOracle.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        ensureNonzeroAddress(accountant);
+        ACCOUNTANT = IAccountant(accountant);
+    }
+
+    /**
+     * @notice Gets the ETH for 1 weETH LRT
+     * @return amount Amount of ETH
+     */
+    function _getUnderlyingAmount() internal view override returns (uint256) {
+        return ACCOUNTANT.getRate();
```
Comment (chechu) [inline contracts/oracles/WeETHAccountantOracle.sol:35]: I would call `getRateSafe`, that takes into account if the Accountant is paused

File: deploy/13-deploy-weETHAccountant-oracle.ts
Comment (chechu) [inline deploy/13-deploy-weETHAccountant-oracle.ts:1]: I would rename this file to `deploy/13-deploy-weETHs-oracle.ts`

File: deploy/13-deploy-weETHAccountant-oracle.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        name: "DefaultProxyAdmin",
+        artifact: defaultProxyAdmin,
+      },
+    },
+    skipIfAlreadyDeployed: true,
+  });
+};
+
+export default func;
+func.tags = ["weETHAccountant"];
```
Comment (chechu) [inline deploy/13-deploy-weETHAccountant-oracle.ts:45]:
```suggestion
func.tags = ["WeETHAccountantOracle_weETHs"];
```

File: deploy/13-dependencies-weETHAccountant-oracle.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    from: deployer,
+    contract: "MockAccountant",
+    args: [],
+    log: true,
+    autoMine: true,
+    skipIfAlreadyDeployed: true,
+  });
+
+  const mockAccountant = await ethers.getContract("MockAccountant_weETHs");
+  await mockAccountant.transferOwnership(proxyOwnerAddress);
```
Comment (chechu) [inline deploy/13-dependencies-weETHAccountant-oracle.ts:23]: Would it make sense to set a rate here in the mock contract, for example 1.1?

File: deploy/13-dependencies-weETHAccountant-oracle.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export default func;
+func.tags = ["weETHAccountant"];
```
Comment (chechu) [inline deploy/13-dependencies-weETHAccountant-oracle.ts:27]:
```suggestion
func.tags = ["weETHAccountant_weETHs"];
```

File: deploy/13-dependencies-weETHAccountant-oracle.ts
Comment (chechu) [inline deploy/13-dependencies-weETHAccountant-oracle.ts:1]: We could rename this file to `deploy/13-dependencies-weETHs-oracle.ts`

File: contracts/oracles/WeETHAccountantOracle.sol [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        address weethLRT,
+        address weth,
+        address resilientOracle
+    ) CorrelatedTokenOracle(weethLRT, weth, resilientOracle) {
+        ensureNonzeroAddress(accountant);
+        ACCOUNTANT = IAccountant(accountant);
+    }
+
+    /**
+     * @notice Gets the ETH for 1 weETH LRT
```
Comment (chechu) [inline contracts/oracles/WeETHAccountantOracle.sol:31]:
```suggestion
     * @notice Gets the WETH for 1 weETH LRT
```

Comment (0xSolDev) [reply contracts/oracles/WeETHAccountantOracle.sol:31]: Fixed

**Review by GitGuru7 — APPROVED** [review]

File: contracts/oracles/WeETHAccountantOracle.sol [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        address weth,
+        address resilientOracle
+    ) CorrelatedTokenOracle(weethLRT, weth, resilientOracle) {
+        ensureNonzeroAddress(accountant);
+        ACCOUNTANT = IAccountant(accountant);
+    }
+
+    /**
+     * @notice Gets the WETH for 1 weETH LRT
+     * @return amount Amount of ETH
```
Comment (chechu) [inline contracts/oracles/WeETHAccountantOracle.sol:32]:
```suggestion
     * @return amount Amount of WETH
```

**Pull request merged**